soroban-sdk-auth/src/lib.rs

```diff
@@ -0,0 +1,112 @@
+#![no_std]
+
+use soroban_sdk::{serde::Serialize, Account, BigInt, BytesN, Env, EnvVal, Symbol, TryIntoVal};
+
+pub mod public_types;
+use crate::public_types::{
+    Identifier, KeyedAccountAuthorization, KeyedAuthorization, KeyedEd25519Signature, Message,
+    MessageV0,
+};
+
+pub trait NonceAuth {
+    fn read_nonce(e: &Env, id: Identifier) -> BigInt;
+    fn read_and_increment_nonce(&self, e: &Env, id: Identifier) -> BigInt;
+    fn get_keyed_auth(&self) -> &KeyedAuthorization;
+}
+
+pub fn check_ed25519_auth(
+    e: &Env,
+    auth: &KeyedEd25519Signature,
+    function: Symbol,
+    parameters: EnvVal,
+) {
+    let msg = MessageV0 {
+        function,
+        contrct_id: e.get_current_contract(),
+        network_id: e.ledger().network_passphrase(),
+        parameters: parameters.to_raw().try_into_val(e).unwrap(),
+    };
+    let msg_bin = Message::V0(msg).serialize(e);
+
+    e.verify_sig_ed25519(
+        auth.public_key.clone().into(),
+        msg_bin,
+        auth.signature.clone().into(),
+    );
+}
+
+pub fn check_account_auth(
+    e: &Env,
+    auth: &KeyedAccountAuthorization,
+    function: Symbol,
```

Comment:
@sisuresh I think you suggested at some point about getting the currently invoked function Symbol from the host. Now seeing this API I think that could make sense for simplifying the API into auth, and reducing chances of human error. Wdyt?

Reply:
I think that would be a nice to have. I'm not sure how this would be done though. Would the `derive_contract_function_set` macro (or a related one) need to be updated to store the current contract call?

Reply:
A host function would be added to get the current invoked host function Symbol. So you'd call that inside this function.

Reply:
Opened an issue for this - https://github.com/stellar/rs-soroban-sdk/issues/477